go/vt/vtgate/engine/join.go

```diff
@@ -218,3 +218,14 @@ func (code JoinOpcode) String() string {
 func (code JoinOpcode) MarshalJSON() ([]byte, error) {
 	return ([]byte)(fmt.Sprintf("\"%s\"", code.String())), nil
 }
+
+func combineVars(bv1, bv2 map[string]*querypb.BindVariable) map[string]*querypb.BindVariable {
+	out := make(map[string]*querypb.BindVariable)
```

Comment:
Not sure this would make a difference, but you can add a second parameter of len(bv1) + len(bv2)

Reply:
I can make this change. It can only be better.